web/oidc/oidc.go

```diff
@@ -308,6 +350,54 @@ func Login(c echo.Context) error {
 	return createSessionAndRedirect(c, inst, redirect, confirm, sid)
 }
 
+func acceptSharing(c echo.Context, ownerInst *instance.Instance, conf *Config, token string, state *stateHolder) error {
+	params, err := getUserInfo(conf, token)
+	if err != nil {
+		return err
+	}
+
+	sub, ok := params["sub"].(string)
+	if !ok {
+		logger.WithNamespace("oidc").Errorf("Missing sub")
+		return ErrAuthenticationFailed
+	}
+	var memberInst *instance.Instance
+	domain, err := extractDomain(conf, params)
+	if err == nil {
+		memberInst, err = lifecycle.GetInstance(domain)
+	} else if err == ErrAuthenticationFailed && conf.AllowCustomInstance {
+		memberInst, err = findInstanceBySub(sub, state.OIDCContext)
+	}
+	if err != nil {
+		return renderError(c, nil, http.StatusNotFound, "Sorry, the cozy was not found.")
+	}
+
+	s, err := sharing.FindSharing(ownerInst, state.SharingID)
```

Comment:
I'm not sure I really understand what's happening here. Why are we looking for the sharing on the owner's instance (which might not even be on the same server) to accept the sharing on the member's side?